ENVOY-LISTENER-EXAMPLE.md

```diff
@@ -0,0 +1,167 @@
+    name: prometheus_route
+    virtual_hosts:
+    - name: "prometheus_metrics_route"
+      domains: ["*"]
+      routes:
+      - match:
+          path: "/metrics"
+        route:
+          cluster: "envoy-admin"
+          prefix_rewrite: "/stats/prometheus"
+```
+
+Note that these Envoy resources are not validated by k8s at all, so
+any errors in the Envoy resources will only be seen by the Cilium
+Agent observing these CRDs. This means that `kubectl apply` will
+report success, while parsing and/or installing the resources to the
+node-local Envoy instance may have failed. Currently the only way of
+verifying this is by observing Cilium Agent logs for errors and
+warnings:
+
+```
+kubectl logs -n kube-system cilium-xxxxx | grep -E "level=(error|warning)"
+```
+
+Please replace `cilium-xxxxx` with one of the Cilium Agent pod names
+you see in `kubectl get pods -A`.
+
+This example contains two Envoy resources, one Listener and one
+RouteConfiguration. Each of the resources needs to have a unique
+name. Envoy resource names for different resource types can be the
+same, but they are still separated by the resource type.
+
+Note that same Envoy resource names for a given resource type in
+different CEC CRDs will overwrite each other. So for now you need make
+sure the resource names are different in different CEC CRDs you may
+deploy. This limitation may be lifted in future versions e.g., by
+namespacing the resource names with the CEC CRD name.
+
+
+## Supported Envoy API Versions
+
+As of now only the Envoy API v3 is supported.
+
+
+## Supported Envoy Extension Resource Types
+
+Envoy extensions are resource types that may or may not be built in to
+an Envoy build. The standard types referred to in Envoy documetation,
+such as `type.googleapis.com/envoy.config.listener.v3.Listener`, and
+`type.googleapis.com/envoy.config.route.v3.RouteConfiguration`, are
+always available.
+
+Cilium nodes deploy an Envoy image to support Cilium HTTP policy
+enforcement and observability. This build of Envoy has been optimized
+for the needs of the Cilium Agent and does not contain many of the
+Envoy extensions available in the Envoy code base.
+
+To see which Envoy extensions are available, please have a look at
+[this Envoy extensions configuration
+file](https://github.com/cilium/proxy/blob/master/envoy_build_config/extensions_build_config.bzl).
+Only the extensions that have not been commented out with `#` are
+built in to the Cilium Envoy image. Currently this contains the
+following extensions:
+
+- `envoy.filters.http.router`
+- `envoy.filters.listener.tls_inspector`
+- `envoy.filters.network.http_connection_manager`
+- `envoy.filters.network.mongo_proxy`
+- `envoy.filters.network.mysql_proxy`
+- `envoy.filters.network.tcp_proxy`
+- `envoy.stat_sinks.metrics_service`
+- `envoy.transport_sockets.raw_buffer`
+
+We will evolve the list of built in extensions based on customer
+feedback for those extensions to be available to be used from CEC
+CRDs.
```

Comment:
This is really useful - I'm thinking we should turn this into a Discussion for getting feedback